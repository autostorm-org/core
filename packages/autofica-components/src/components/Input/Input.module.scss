.root {
  background: var(--input-background-color) 0% 0% no-repeat padding-box;
  border: 1px solid var(--input-border-color);
  border-radius: 4px;
  opacity: 1;
  display: flex;
  width: 100%;
  padding: 10px 15px 10px 15px;
  box-sizing: border-box;
  align-items: center;
  transition: 0.3s ease;
  transition-property: color, background-color, width, height, margin, padding,
    border, box-shadow;
  &:focus-within {
    box-shadow: 0 0 0 0.2rem var(--input-border-color-focus, #b3b3b3);
    outline: none;
    border: 1px solid transparent;
  }
  & svg {
    color: var(--input-color);
  }
}

.input {
  border-width: 0px;
  border: none;
  width: 100%;
  text-align: left;
  font-size: 14px;
  font-weight: var(--input-font-weight);
  font-family: var(--input-font-family);
  letter-spacing: 0;
  background-color: inherit;
  color: var(--input-color);
  &:focus {
    outline: none;
  }

  &::-webkit-calendar-picker-indicator {
    filter: var(--input-calendar-picker-indicator-filter);
  }
  &:focus::-webkit-calendar-picker-indicator {
    filter: var(--input-calendar-picker-indicator-filter-focus);
  }
}
