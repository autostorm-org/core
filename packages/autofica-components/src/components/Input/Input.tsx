import React from "react";
import styles from "./Input.module.scss";
import type {
  IOverridable,
  IStylable,
  IAriaCombobox,
  IAriaLabel,
} from "../../types";

type InputSupportedType =
  | "text"
  | "number"
  | "file"
  | "email"
  | "date"
  | "tel"
  | "password";

interface InputProps
  extends IOverridable,
    IStylable,
    IAriaCombobox,
    IAriaLabel {
  iconLeft?: React.JSXElementConstructor<{}>;
  iconRight?: React.JSXElementConstructor<{}>;
  placeholder?: string;
  type?: InputSupportedType;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onFocus?: React.FocusEventHandler<HTMLInputElement>;
  onFocusCapture?: React.FocusEventHandler<HTMLInputElement>;
  onBlur?: React.FocusEventHandler<HTMLInputElement>;
  id: string;
  autocomplete?: string;
  value?: string;
  role?: string;
}
const Input = React.forwardRef<HTMLLabelElement, InputProps>((props, ref) => {
  return (
    <label
      className={`${styles.root} ${props.override || ""}`}
      style={props.style}
      htmlFor={props.id}
      ref={ref}
      aria-label={props["aria-label"]}
    >
      {props.iconLeft && <props.iconLeft />}
      <input
        className={styles.input}
        type={props.type}
        placeholder={props.placeholder}
        value={props.value}
        onChange={props.onChange}
        onFocusCapture={props.onFocusCapture}
        onFocus={props.onFocus}
        onBlur={props.onBlur}
        autoComplete={props.autocomplete}
        id={props.id}
        role={props.role}
        aria-autocomplete={props["aria-autocomplete"]}
        aria-haspopup={props["aria-haspopup"]}
        aria-owns={props["aria-owns"]}
        aria-expanded={props["aria-expanded"]}
        aria-activedescendant={props["aria-activedescendant"]}
      />
      {props.iconRight && <props.iconRight />}
    </label>
  );
});
export default React.memo(Input);
export type { InputProps, InputSupportedType };
