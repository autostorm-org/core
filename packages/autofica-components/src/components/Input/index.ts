// Input
export { default as Input } from "./Input";
export type { InputProps, InputSupportedType } from "./Input";
