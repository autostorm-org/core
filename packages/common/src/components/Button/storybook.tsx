import React from "react";
import Button from "./Button";

// This default export determines where you story goes in the story list
export default {
  title: "Button",
  component: Button,
};

export const ButtonStory = () => {
  return <Button>Hello</Button>;
};
