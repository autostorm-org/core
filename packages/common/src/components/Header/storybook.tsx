import React from "react";
import Header from "./Header";
import HeaderLogo from "./HeaderLogo";
import { HeaderLocations } from "./constants";
import HeaderItem from "./HeaderItem";
import Spain from "../../assets/flags/Spain.png";
import Brazil from "../../assets/flags/Brazil.png";
import ResponsiveTemplate from "../ResponsiveTemplate";
import ListFixed from "../ListFixed";
import Icon from "../Icon";
import HeaderItemAbsolute from "./HeaderItemAbsolute";
import { ListFullScreen, ListFullScreenRenderer } from "../List";
import { Menu } from "react-feather";
// This default export determines where you story goes in the story list
export default {
  title: "Header",
  component: Header,
};
declare var require: any;

const logoPath = "./../../assets/autofica-defi-day-logo.svg";

const Desktop = (args) => (
  <Header>
    <HeaderLogo
      logoPath={require("./../../assets/autofica-defi-logo-short-6.svg")}
      itemLocation={HeaderLocations.left}
    />
    <HeaderItem>Loan Manager</HeaderItem>
    <HeaderItem>Whitepaper</HeaderItem>
    <HeaderItem
      dropDown
      sections={[
        {
          links: [
            {
              content: (
                <>
                  <span>br-PT</span>{" "}
                  <Icon
                    src={Brazil}
                    width={23}
                    height={13}
                    alt={"portuguese"}
                  />
                </>
              ),
            },
          ],
        },
      ]}
      Submenu={ListFixed}
    >
      es-ES <Icon src={Spain} width={23} height={13} alt={"espanol"} />
    </HeaderItem>
  </Header>
);

const Mobile = (args) => {
  const [isOpen, setOpen] = React.useState(false);
  return (
    <Header
      submenu={
        isOpen ? (
          <ListFullScreen>
            <ListFullScreenRenderer>Vende</ListFullScreenRenderer>{" "}
            <ListFullScreenRenderer>Contacto</ListFullScreenRenderer>
            <ListFullScreenRenderer>Ayuda</ListFullScreenRenderer>
            <ListFullScreenRenderer>Mi Cuenta</ListFullScreenRenderer>
          </ListFullScreen>
        ) : null
      }
    >
      <HeaderLogo
        logoPath={require("./../../assets/autofica-defi-day-logo.svg")}
        itemLocation={HeaderLocations.center}
        overrides={{ paddingTop: 10, paddingBottom: 10, width: 125 }}
      />
      <HeaderItemAbsolute
        itemLocation={HeaderLocations.right}
        onClick={function () {
          setOpen(!isOpen);
        }}
      >
        <Menu color="white" />
      </HeaderItemAbsolute>
    </Header>
  );
};

const ResponsiveStory = () => (
  <ResponsiveTemplate mobile={Mobile} tablet={Desktop} desktop={Desktop} />
);

export const FirstStory = ResponsiveStory.bind({});

FirstStory.args = {
  /* the args you need here will depend on your component */
};
