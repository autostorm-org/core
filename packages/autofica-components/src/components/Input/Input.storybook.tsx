import React from "react";
import Input from "./Input";
import { Typography, TypographyVariant_ENUM } from "../Typography";
// This default export determines where you story goes in the story list
export default {
  title: "Input",
  component: Input,
};

const Template = () => {
  return (
    <div style={{ padding: 10 }}>
      <Typography variant={TypographyVariant_ENUM.h6}>Text input</Typography>
      <Input
        id={"text-input"}
        type={"text"}
        placeholder={"Text input"}
        autocomplete={"off"}
      />
      <Typography variant={TypographyVariant_ENUM.h6}>Number input</Typography>
      <Input id={"number-input"} type={"number"} placeholder={"Number input"} />
      <Typography variant={TypographyVariant_ENUM.h6}>File input</Typography>
      <Input id={"file-input"} type={"file"} placeholder={"File input"} />
      <Typography variant={TypographyVariant_ENUM.h6}>Email input</Typography>
      <Input id={"email-input"} type={"email"} placeholder={"Email input"} />
      <Typography variant={TypographyVariant_ENUM.h6}>Date input</Typography>
      <Input id={"date-input"} type={"date"} placeholder={"Date input"} />
      <Typography variant={TypographyVariant_ENUM.h6}>Tel input</Typography>
      <Input id={"tel-input"} type={"tel"} placeholder={"Tel input"} />
      <Typography variant={TypographyVariant_ENUM.h6}>
        Password input
      </Typography>
      <Input id={"password-input"} type={"password"} />
    </div>
  );
};

export const FirstStory = Template.bind({});
