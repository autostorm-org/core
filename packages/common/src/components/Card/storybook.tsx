import React from "react";
import Card from "./Card";
import Typography, { EnumTypographyVariant } from "../Typography";
import Button from "../Button";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import { ThemeContext, nightTheme, defaultTheme } from "../../theme";
import ResponsiveTemplate from "../ResponsiveTemplate";
// This default export determines where you story goes in the story list
export default {
  title: "Card",
  component: Card,
  decorators: [withKnobs],
};

const MobileCard = () => {
  return (
    <div style={{ height: 184, width: 350 }}>
      <Card>
        <Typography variant={EnumTypographyVariant.h3}>Card title</Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Card body contents. Card body contents. Card body contents. Card body
          contents.
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 9 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const DesktopCard = () => {
  return (
    <div style={{ height: 240, width: 560 }}>
      <Card style={{ padding: "30px" }}>
        <Typography variant={EnumTypographyVariant.h1}>
          Crypto car loan finder
        </Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Get a collateralized loan for your next car purchase. Chose your
          backing assets. Powered by Multiplier.{" "}
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 10 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const ResponsiveStory = () => {
  const isDayMode = boolean("DAY MODE", true);
  return (
    <ThemeContext.Provider value={isDayMode ? defaultTheme : nightTheme}>
      <ResponsiveTemplate
        mobile={MobileCard}
        desktop={DesktopCard}
        tablet={DesktopCard}
      />
    </ThemeContext.Provider>
  );
};

export const FirstStory = ResponsiveStory.bind({});
