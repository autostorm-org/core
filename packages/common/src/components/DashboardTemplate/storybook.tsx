import React from "react";
import DashboardTemplate from "./DashboardTemplate";
import { BoundedRow } from "../BoundedRow";
import { Typography, TypographyVariant_ENUM } from "../Typography";
import Icon from "../Icon";
import Spain from "../../assets/flags/Spain.png";
import { HeaderResponsive } from "../Header";
import DashboardOptionsColumn from "./DashboardOptionsColumn";
import DashboardOptionsItem from "./DashboardOptionsItem";
import { PillButton } from "./../PillButton";
import { Eye, ArrowLeftCircle, DollarSign, ArrowUp, Key } from "react-feather";

export default {
  title: "Dashboard Template Stories",
};

export const DashboardOptionsColumnStory = () => {
  return (
    <DashboardOptionsColumn>
      <DashboardOptionsItem activeElement href={"#"}>
        hello now
      </DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
    </DashboardOptionsColumn>
  );
};

export const DashboardTemplateStory = () => {
  return (
    <>
      <HeaderResponsive
        options={[
          { content: "Loan Manager", href: "#" },
          { content: "Whitepaper", href: "#" },
          {
            content: (
              <>
                es-ES{" "}
                <Icon src={Spain} width={23} height={13} alt={"espanol"} />
              </>
            ),
          },
        ]}
      />
      <div style={{ marginTop: 30 }}>
        <BoundedRow>
          <DashboardTemplate
            mainContent={
              <Typography
                variant={TypographyVariant_ENUM.h3}
                style={{ padding: 30 }}
              >
                Main Content
              </Typography>
            }
            optionsContent={
              <>
                <DashboardOptionsColumn style={{ borderTopLeftRadius: 8 }}>
                  <div style={{ marginTop: 30, marginBottom: 20 }}>
                    <PillButton
                      onClick={() => {
                        console.log("CLIICK");
                      }}
                    >
                      <Icon
                        src={"crypto-logos/bsc.svg"}
                        height={22}
                        width={22}
                        alt={"BSC logo"}
                        style={{ marginRight: 10 }}
                      />
                      BSC Mainnet
                    </PillButton>
                  </div>

                  <DashboardOptionsItem href={"#"} activeElement>
                    <Eye style={{ marginRight: 10 }} />
                    Loans Overview
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowLeftCircle style={{ marginRight: 10 }} />
                    Borrow
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <DollarSign style={{ marginRight: 10 }} />
                    Payback
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowUp style={{ marginRight: 10 }} />
                    Withdraw Collateral
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <Key style={{ marginRight: 10 }} />
                    Deposit
                  </DashboardOptionsItem>
                </DashboardOptionsColumn>
              </>
            }
          />
        </BoundedRow>
      </div>
    </>
  );
};
