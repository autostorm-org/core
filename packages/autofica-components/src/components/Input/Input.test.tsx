import React from "react";
import renderer from "react-test-renderer";
import { FirstStory } from "./Input.storybook";
describe("Input Test", () => {
  it("Should render Input component", () => {
    const root = renderer.create(<FirstStory />);
    let tree = root.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
