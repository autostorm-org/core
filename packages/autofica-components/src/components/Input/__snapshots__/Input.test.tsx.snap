// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Test Should render Input component 1`] = `
<div
  style={
    Object {
      "padding": 10,
    }
  }
>
  <span
    className="h6"
  >
    Text input
  </span>
  <label
    className="root "
    htmlFor="text-input"
  >
    <input
      autoComplete="off"
      className="input"
      id="text-input"
      placeholder="Text input"
      type="text"
    />
  </label>
  <span
    className="h6"
  >
    Number input
  </span>
  <label
    className="root "
    htmlFor="number-input"
  >
    <input
      className="input"
      id="number-input"
      placeholder="Number input"
      type="number"
    />
  </label>
  <span
    className="h6"
  >
    File input
  </span>
  <label
    className="root "
    htmlFor="file-input"
  >
    <input
      className="input"
      id="file-input"
      placeholder="File input"
      type="file"
    />
  </label>
  <span
    className="h6"
  >
    Email input
  </span>
  <label
    className="root "
    htmlFor="email-input"
  >
    <input
      className="input"
      id="email-input"
      placeholder="Email input"
      type="email"
    />
  </label>
  <span
    className="h6"
  >
    Date input
  </span>
  <label
    className="root "
    htmlFor="date-input"
  >
    <input
      className="input"
      id="date-input"
      placeholder="Date input"
      type="date"
    />
  </label>
  <span
    className="h6"
  >
    Tel input
  </span>
  <label
    className="root "
    htmlFor="tel-input"
  >
    <input
      className="input"
      id="tel-input"
      placeholder="Tel input"
      type="tel"
    />
  </label>
  <span
    className="h6"
  >
    Password input
  </span>
  <label
    className="root "
    htmlFor="password-input"
  >
    <input
      className="input"
      id="password-input"
      type="password"
    />
  </label>
</div>
`;
